src/imap_mag/cli/fetchScience.py

```diff
@@ -0,0 +1,74 @@
+"""Program to retrieve and process MAG CDF files."""
+
+import typing
+from pathlib import Path
+
+import pandas as pd
+import typing_extensions
+
+from .. import appUtils
+from ..client.sdcDataAccess import ISDCDataAccess
+
+
+class FetchScienceOptions(typing.TypedDict):
+    """Options for SOC interactions."""
+
+    level: str
+    start_date: str
+    end_date: str
+    output_dir: str
+
+
+class FetchScience:
+    """Manage SOC data."""
+
+    __descriptors: list[str]
+    __flavors: list[str]
+
+    __data_access: ISDCDataAccess
+
+    def __init__(
+        self,
+        data_access: ISDCDataAccess,
+        descriptors: list[str] = ["norm", "burst"],
+        flavors: list[str] = ["-magi", "-mago"],
+    ) -> None:
+        """Initialize SDC interface."""
+
+        self.__data_access = data_access
+        self.__descriptors = descriptors
+        self.__flavors = flavors
+
+    def download_latest_science(
+        self, **options: typing_extensions.Unpack[FetchScienceOptions]
+    ) -> list[Path]:
+        """Retrieve SDC data."""
+
+        downloaded = []
+
+        for descriptor in self.__descriptors:
+            date_range: pd.DatetimeIndex = pd.date_range(
```

Comment:
This was from the other code but why do we use pandas here? Is it just for doing date ranges?

Reply:
Yes, that was the best function I found

Reply:
Feels like a big dependency to add for pretty simple utility. We have not really had the pandas vs numby vs both discussion but would be tempted to say we should not pull in all of pandas just for this. There might be some code you can inline from https://github.com/pandas-dev/pandas/blob/main/pandas/core/indexes/datetimes.py#L130 instead but I think I would just use list comprehension:

```
start = datetime.datetime(2024,8,1)
end = datetime.datetime(2024,8,7)
[end - datetime.timedelta(days=x) for x in range((end-start).days+1)]

# returns array. the +1 is inclusive of start and end:
[datetime.datetime(2024, 8, 7, 0, 0), datetime.datetime(2024, 8, 6, 0, 0), datetime.datetime(2024, 8, 5, 0, 0), datetime.datetime(2024, 8, 4, 0, 0), datetime.datetime(2024, 8, 3, 0, 0), datetime.datetime(2024, 8, 2, 0, 0), datetime.datetime(2024, 8, 1, 0, 0)]

```

Reply:
I also use `pandas` for converting a string to a `datetime` ([`pandas.to_datetime`](https://pandas.pydata.org/docs/reference/api/pandas.to_datetime.html)), as it does not require a specific input format for the date/time, but is flexible and can interpret all sorts of different formats. I think this function would be harder to replace than `pandas.date_range`, unless we enforce a (few?) specific format(s). 

I am not familiar with the differences between `pandas` and `numpy`, so I have no preference. I've had a quick try at removing `pandas` from our `pyproject.toml` dependencies, and we still end up pulling it in in our lock file, as some of our dependencies (including `numpy`??) depend on it...

Reply:
Pandas depends on numpy, not the other way around, Maybe have a look at dateutil?